### PR TITLE
[SD-2729] geo block the ipnews output

### PR DESCRIPTION
**Pull request by marwoodandrew**

### CODE REVIEW

**Comment by akintolga** [comment]
:+1:

**Comment by ioanpocol** [comment]
:+1:

File: server/apps/publish/formatters/aap_ipnews_formatter_test.py [hunk 1]
```diff
@@ -139,3 +140,65 @@ def TestMultipleCategories(self):
...
+        with self.app.app_context():
+            subscriber = self.app.data.find('subscribers', None, None)[0]
+
+            f = AAPIpNewsFormatter()
+            seq, doc = f.format(article, subscriber)[0]
+            codes = set(doc['selector_codes'].split(' '))
+            expected_codes_str = 'an5 an4 an7 an6 ax5 ax6 ax7 an8 px6 ax4 ax8 px5 0ah 0px'
+            expected_codes_str += ' px8 0fh px7 px4 pn4 pn5 pn6 pn7 px0'
+            expected_codes = set(expected_codes_str.split(' '))
+            self.assertSetEqual(codes, expected_codes)
```
Comment (plamut) [inline server/apps/publish/formatters/aap_ipnews_formatter_test.py]: Sidenote: `self.assertEqual()` already knows which type-specific method for equality comparison to call, thus it's usually not necessary to call a specific method directly.

(still no need to change this, just a remark)

File: server/apps/publish/formatters/field_mappers/selectorcode_mapper.py [hunk 2]
```diff
@@ -82,92 +140,146 @@ def map(self, article, category, **kwargs):
...
 
+    def _get_geo_abbreviation(self, name):
+        """
+        passed the name that is kept in the article uses it to find the value from the vocabulary
+        :param name:
+        :return: the abbreviatoin for the geographical restriction
+        """
+        geo_resource = superdesk.get_resource_service('vocabularies').find_one(req=None,
+                                                                               _id='geographical_restrictions')
+        item = [i for i in geo_resource.get('items') if i['name'] == name][0]
```
Comment (plamut) [inline server/apps/publish/formatters/field_mappers/selectorcode_mapper.py]: Depending on the number of items, using an iterator can be quite a bit faster than constructing a whole list.
```py
item = next(i for i in geo_resource.get('items') if i['name'] == name, None)
```

File: server/apps/publish/formatters/aap_ipnews_formatter_test.py [hunk 3]
```diff
@@ -139,3 +140,65 @@ def TestMultipleCategories(self):
...
+
+            f = AAPIpNewsFormatter()
+            seq, doc = f.format(article, subscriber)[0]
+            codes = set(doc['selector_codes'].split(' '))
+            expected_codes_str = 'an5 an4 an7 an6 ax5 an3 ax6 ax7 0hw an8 0ir px6 ax4 ax3 ax8 px5 0ah 0px px3 az'
+            expected_codes_str += ' px8 0fh px7 px4 pn3 pn4 pn5 pn6 pn7 px0'
+            expected_codes = set(expected_codes_str.split(' '))
+            self.assertSetEqual(codes, expected_codes)
+
+    def TestGeoBlockNotTwoStates(self):
```
Comment (plamut) [inline server/apps/publish/formatters/aap_ipnews_formatter_test.py]: For consistency, I would suggest renaming these two test methods to start with lowercase (`Test...` --> `test...`).

**Comment by plamut** [comment]
:+1: 

Looks good in general, apart for some smaller remarks.

**Comment by amagdas** [comment]
:+1:

File: server/apps/publish/formatters/field_mappers/selectorcode_mapper.py (hunk 2 above)
Comment (marwoodandrew) [inline server/apps/publish/formatters/field_mappers/selectorcode_mapper.py]: OK done!

File: server/apps/publish/formatters/aap_ipnews_formatter_test.py (hunk 3 above)
Comment (marwoodandrew) [inline server/apps/publish/formatters/aap_ipnews_formatter_test.py]: I think I've done this in some other other test modules as well, will fix 'em as I come across them

**Pull request merged**